packages/datagrid/src/components/DataGrid/DataGrid.component.js

```diff
@@ -80,6 +80,30 @@ export function getRowDataInfos(rowData) {
 	);
 }
 
+/**
+ * redrawRows - call redrawRows only if necessary. (improve ag-grid performance)
+ *
+ * @param  {object} props     component props
+ * @param  {object} gridAPI   ag-grid api
+ * @param  {object} prevProps previous component prop
+ */
+export function redrawRows(props, gridAPI, prevProps) {
+	if (props.loading || !gridAPI) {
+		return;
+	}
+
+	const prevInfos = getRowDataInfos(prevProps.rowData);
+	const currentInfos = getRowDataInfos(props.rowData);
+
+	if (
+		prevInfos.loaded !== currentInfos.loaded ||
```

Comment:
not a fan of using many boolean for loading status, this can lead to impossible state easily, like having none of them true, or many of them true like, loaded and notLoaded, i rather have a `status` and `statusList` with the four status and a checker when updating `status` value